addons/account/models/product.py

```diff
@@ -114,6 +114,35 @@ def _check_uom_not_in_invoice(self):
                 "If you want to change its Unit of Measure, please archive this product and create a new one."
             ))
 
+    def _get_default_company_taxes(self, companies):
+        default_taxes_ids = [company.account_sale_tax_id.id for company in companies if company.account_sale_tax_id]
+        default_supplier_taxes_ids = [company.account_purchase_tax_id.id for company in companies if company.account_purchase_tax_id]
+
+        return default_taxes_ids, default_supplier_taxes_ids
```

Comment:
Although, I like it (a lot) personally. I'm not sure we need to create a function for that as it is only use once (if I'm not mistaken).

Curious to see what @qdp-odoo thinks.